src/lib/builtins/array.rs

```diff
@@ -85,11 +85,19 @@ pub fn make_array(this: &Value, args: &[Value], ctx: &mut Interpreter) -> Result
     // between indices and values): this creates an Object with no prototype
 
     // Create length
-    let length = Property::new()
-        .value(to_value(args.len() as i32))
-        .writable(true)
-        .configurable(false)
-        .enumerable(false);
+    let length = if args.len() > 1 {
```

Comment:
You can extract just the length and write the logic for creating the property once, e.g.
```rust
let n = if args.len() > 1 { args.len() } ...
```